connor/traderModule.go

```diff
@@ -0,0 +1,539 @@
+			if err != nil {
+				return fmt.Errorf("cannot get deal from Market %v\r\n", err)
+			}
+			deal := getDealFromMarket.Deal
+
+			log.Printf("Deploying NEW CONTAINER ==> for dealDB %v (deal on market: %v)\r\n", dealDb.DealID, deal.GetId().Unwrap().String())
+			task, err := t.pool.DeployNewContainer(ctx, t.c.cfg, deal, image)
+			if err != nil {
+				t.c.db.UpdateDealInDB(deal.Id.Unwrap().Int64(), int32(DeployStatusNOTDEPLOYED))
+				return fmt.Errorf("Cannot deploy new container from task %s\r\n", err)
+			} else {
+				t.c.db.UpdateDealInDB(deal.Id.Unwrap().Int64(), int32(DeployStatusDEPLOYED))
+				fmt.Printf("New deployed task %v, for deal ID: %v\r\n", task.GetId(), deal.GetId().String())
+			}
+
+			bidOrder, err := t.c.Market.GetOrderByID(ctx, &sonm.ID{Id: deal.GetBidID().String()})
+			if err != nil {
+				fmt.Printf("Cannot get order by Id: %v\r\n", err)
+				return err
+			}
+			bench, err := t.GetBidBenchmarks(bidOrder)
+			if err != nil {
+				fmt.Printf("Cannot get benchmarks from bid Order : %v\r\n", bidOrder.Id.Unwrap().Int64())
+				return err
+			}
+			t.ReinvoiceOrder(ctx, t.c.cfg, &sonm.Price{PerSecond: deal.GetPrice()}, bench, "Reinvoice(active deal)")
+		}
+	}
+	return nil
+}
+
+func (t *TraderModule) GetChangeRequest(ctx context.Context, dealChangeRequest *sonm.DealInfoReply) error {
+	time.Sleep(time.Duration(t.c.cfg.Sensitivity.WaitingTimeCRSec))
+	requestsList, err := t.c.DealClient.ChangeRequestsList(ctx, dealChangeRequest.Deal.Id)
+	if err != nil {
+		return err
+	}
+	for _, cr := range requestsList.Requests {
+		if cr.DealID == dealChangeRequest.Deal.Id && cr.Status == sonm.ChangeRequestStatus_REQUEST_REJECTED {
+			t.c.DealClient.Finish(ctx, &sonm.DealFinishRequest{
+				Id: dealChangeRequest.Deal.Id,
+			})
+		}
+	}
+	return nil
+
+}
+
+func (t *TraderModule) ClonePrice(def *big.Int) (*big.Int, error) {
+	clone := big.NewInt(def.Int64())
+	return big.NewInt(0).Set(clone), nil
+}
+
+func (t *TraderModule) GetChangePercent(actualPriceForPack *big.Int, dealPrice *big.Int) (float64, error) {
+	newClone, err := t.ClonePrice(actualPriceForPack)
+	if err != nil {
+		return 0, fmt.Errorf("cannot get clone price %v", err)
+	}
+	dealClone, err := t.ClonePrice(dealPrice)
+	if err != nil {
+		return 0, fmt.Errorf("cannot get clone price %v", err)
+	}
+	div := big.NewFloat(params.Ether)
+
+	vNew := big.NewFloat(0).SetInt(newClone)
+	rNew := big.NewFloat(0).Quo(vNew, div)
+
+	vOld := big.NewFloat(0).SetInt(dealClone)
+	rOld := big.NewFloat(0).Quo(vOld, div)
+
+	fOld, _ := rOld.Float64()
+	fnew, _ := rNew.Float64()
+	delta := (fnew * 100) / fOld
+	return delta, nil
+}
+func (t *TraderModule) SaveActiveDealsIntoDB(ctx context.Context, dealCli sonm.DealManagementClient) error {
+	getDeals, err := dealCli.List(ctx, &sonm.Count{Count: 100})
+	if err != nil {
+		fmt.Printf("Cannot get Deals list %v\r\n", err)
+		return err
+	}
+	deals := getDeals.Deal
+	if len(deals) > 0 {
```

Comment:
you can drop this check, if deals is an empty slice - there is no iterations will be performed.